02-wrangling.Rmd

```diff
@@ -304,6 +317,76 @@ From this visualization, we can see that in Calgary, Edmonton, Toronto and Vanco
+
+Chapter 1 discussed two `tidyverse` functions: `select` and `filter` to select a single column and filter rows from a data frame. We will now learn more ways we can use these functions. Recall, the `select` function creates a subset of the *columns* of a data frame, while the `filter` subsets *rows* with specific values.
+
+### Using `select` to extract multiple columns
+
+We can use `select` to obtain a subset of the data frame with multiple columns. The first argument is the name of the data frame. Then we list all the columns we want as arguments separated by commas. Here we create a subset of three columns: language, mother tongue, and language spoken most often at home.
+
+```{r}
+three_columns <- select(can_lang, language, mother_tongue, most_at_home)
+three_columns
+```
+
+
+### Using `select` to extract a range of columns
+
+We can also use `select` to obtain a subset of the data frame constructed from a range of columns. To do this, we use the colon (`:`) operator to denote the range. For example, to get all the columns in the `can_lang` data frame from `language` to `most_at_home` we pass `language:most_at_home` as the second argument to the `select` function.
+
+```{r}
+column_range <- select(can_lang, language:most_at_home)
+column_range
+```
+
+### Using `filter` to extract a single row
+
+Recall we can use the `filter` function to obtain the subset of rows with desired values from a data frame. Again, our first argument is the name of the data frame object, `can_lang`. The second argument is a logical statement to use when filtering the rows. Here, for example, we'll say that we are interested in rows where the language is Mandarin. We use the *equivalency operator* (`==`) to compare the values of the `language` column with the value `"Mandarin"`. With these arguments, `filter` returns a data frame with all the columns of the input data frame but only the rows we asked for in our logical filter statement. 
+
+```{r}
+mandarin <- filter(can_lang, language == "Mandarin")
+mandarin
+```
+
+
+```{r changing_the_units, include = FALSE}
+mandarin_mother_tongue <- mandarin %>%
+  pull(mother_tongue)
+census_popn <- 35151728
+```
+### Using `filter` to extract rows with values above or below a threshold
+
+If we are interested in finding the languages with higher numbers of people who speak it as their mother tongue than Mandarin---which is  `r format(mandarin_mother_tongue, scientific = FALSE, big.mark = ",")` people ---then we can create a filter to obtain rows where the value of `mother_tongue` is greater than `r format(mandarin_mother_tongue, scientific = FALSE, big.mark = ",")`. In this case, we see that `filter` returns a data frame with two rows; this indicates that two languages are spoken more often as their mother tongue compared to Mandarin.
+
+```{r}
+spoke_often_at_home <- filter(can_lang, mother_tongue > 592040)
+spoke_often_at_home
+```
+
+
+### Using `mutate` to add columns or overwrite existing columns
```

Comment:
Oh, now I see there is a section on `mutate`... Can we somehow combine these? Maybe move this up earlier? We could still do the context thing? Perhaps for the 5 cities for the bar charts? To do that we would need `c()` I think to create a vector of the population of the 5 cities... But I think we should introduce `c()` somewhere anyways? Maybe in the vector section? It is so useful - more so than `data.frame()` or `tibble()`.